### PR TITLE
No search results implementation 

### LINKED ISSUE #5423: PNI - implement "no result" state for search
- [x] Implement "no results" state for search
- [x] Hide the "submit a product" block (see linked mockup below) when search is on "no results" state 

---

###  Design -> Dev Handoff Checklist

- [x] Design has been finalized
- [x] Link to related ticket: #5314
- [x] Link to Abstract (make sure it can be viewed by anyone)
- [x] Design changes been merged into the master Sketch file
  - Merged into master Sketch file via Abstract. (_Master > Buyers Guide.sketch > [Home Page Updates (Q4 2020](https://share.goabstract.com/07a9aca5-3903-4f00-828a-6ac8bebb5be3?sha=e49d77cc9da6a70e8c412cd985e28138a3091cd5))_).
- [x] Desktop and mobile mockups are included
  - [[Desktop] Homepage - Category and Search Complete](https://share.goabstract.com/e9fd07c9-605e-4ca9-b99c-7a3686de51cb?collectionLayerId=b4beb7d2-7d3e-4c5a-99ba-51530058b813&mode=design) (no results page)
  - [[Mobile] Homepage - Search No Results](https://share.goabstract.com/e9fd07c9-605e-4ca9-b99c-7a3686de51cb?collectionLayerId=a2e4e32f-6e2d-452b-9fb3-326c30f282fa&mode=design)
- [x] Name of the Sketch file (and specific page/section to look at, if applies)
- [x] Assets are included. Link: https://drive.google.com/open?id=19j7fCYjv-bRgkUlbTBxi1VEEWtedCMdx&authuser=sabrina%40mozillafoundation.org&usp=drive_fs
- [x] Other states for elements are designed: `All` links back to All cleared of search/checkbox

### PR DESCRIPTION
**Pull request by KalobTaulien**
Closes #5423 

- [x] Svg minified with svgo 
- [x] Hide "Submit a product" on category pages when there are no search results 

https://foundation-s-5423-searc-awd7sh.herokuapp.com/en/privacynotincluded/

### CODE REVIEW

**Review by mmmavis — CHANGES_REQUESTED** [review]
Few things need to be addressed:

- missing "Trying to go back..." section
- the "submit a product" block should be hidden (see 2nd checkbox in the ticket #5423)
- resized the "oh no" face so it matches design
- font style of "Oh no! We didn’t find that here." needs adjustment to match design

**Comment by KalobTaulien** [comment]
@mmmavis Thank you for reviewing this! I fixed 2 things, one thing doesnt 

- [x] missing "Trying to go back..." section **Fixed**
- [x] the "submit a product" block should be hidden (see 2nd checkbox in the ticket #5423) **Fixed**
- [x] resize the "oh no" face so it matches design **Fixed (mobile needed to be smaller)**
- [ ] font style of "Oh no! We didn’t find that here." needs adjustment to match design

> font style of "Oh no! We didn’t find that here." needs adjustment to match design

☝️ It does match, unless I'm using the wrong design view? I'm using [this one](https://app.abstract.com/projects/fa246750-1c19-11e8-b2bf-9db5acc6ed5d/branches/master/commits/e49d77cc9da6a70e8c412cd985e28138a3091cd5/files/DBF82218-05B7-47F9-AD7C-8ABDFD109060/layers/507ACA61-C888-4003-B0C7-C71964C7E83E?collectionId=2d92eaf9-cfcb-4aa9-916d-d1a456f6e2cf&collectionLayerId=b4beb7d2-7d3e-4c5a-99ba-51530058b813&mode=build&selected=root-975D73AF-C136-4552-95E8-EFBA827B039D). On mobile the font size gets a bit smaller (the design font size doesnt change), but I asked @sabrinang if this correct and if we should overwrite the `.h3-heading` styleguide behaviour on mobile.

File: network-api/networkapi/buyersguide/templates/fragments/no_search_results.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+<div class="container">
+<img class="mb-3 oh-no-face" src="{% static '_images/buyers-guide/faces/oh-no.svg' %}" alt="{% trans "Sad face. No products were found." %}">
+
+<h3 class="h3-heading">
+  {% trans 'Oh no! We didn’t find that here.' %}
+</h3>
+
+<p class="mb-0">
+  {% url 'buyersguide-home' as home_url %}
+  {% blocktrans %}
```
Comment (mmmavis) [inline network-api/networkapi/buyersguide/templates/fragments/no_search_results.html:12]:
```suggestion
  {% blocktrans trimmed %}
```

File: network-api/networkapi/buyersguide/templates/fragments/no_search_results.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    Try going back to <a href="{{ home_url }}">All.</a> You might find it there.
```
Comment (mmmavis) [inline network-api/networkapi/buyersguide/templates/fragments/no_search_results.html:13]:
```suggestion
    Try going back to <a href="{{ home_url }}">All</a>. You might find it there.
```

**Review by mmmavis — CHANGES_REQUESTED** [review]
2 small fixes required.

Vertical spacing around the "no search results" block needs some tweaks but those can be addressed once this PR has been merged into `search-filter` (in case `search-filter` has them covered already.

File: network-api/networkapi/buyersguide/templates/fragments/no_search_results.html (hunk 1 above)
Comment (KalobTaulien) [reply network-api/networkapi/buyersguide/templates/fragments/no_search_results.html:12]: Ohh good catch!!

**Review by mmmavis — APPROVED** [review]
Awesome! 🎉

**Pull request merged**